src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -144,7 +147,27 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
             TryStartEditing(connection);
         }
 
-        public Task TestConnectionAsync(IConnectionViewModel connection) => Task.CompletedTask;
+        public async Task TestConnectionAsync(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            connection.IsTestingConnection = true;
+            connection.IsTestConnectionSucceeded = false;
+
+            try {
+                await _connectionManager.TestConnectionAsync(connection);
```

Comment:
No, previous test will be canceled (though they can be concurrent)

---
In reply to: [79271393](https://github.com/Microsoft/RTVS/pull/2342#discussion_r79271393) [](ancestors = 79271393)